validphys2/src/validphys/process_options.py

```diff
@@ -229,6 +229,18 @@ def _dyncpt_xq2map(kin_info):
     return x, q2
 
 
+def _dyncpt_xq2map(kin_dict):
+    """Compute x and q2 mapping for DY Z -> ll + jet process.
+    Here pT refers to the transverse momentum of the Z boson.
+    """
+    pT = kin_dict[_Vars.pT]
+    m_Z2 = kin_dict[_Vars.m_Z2]
+    sqrts = kin_dict[_Vars.sqrts]
+    x = (np.sqrt(m_Z2 + pT*pT) + pT) / sqrts
+    q2 = m_Z2 + pT * pT
+    return x, q2
```

Comment:
This is no longer necessary since it is already in master (note that the same function is defined twice)

Reply:
Yes you're right, but the two are slightly different. Should I go for the one implemented by Mark?

Reply:
Oh, I thought they were the same.

No, yours is more correct. The scale used in ZpT is ET (which is what you implemented) while mark's version uses mZ (which is what Z rapidity datasets use). Of course eT -> mZ in the limit of 0 pT

```
    x = (np.sqrt(m_Z2 + pT*pT) + pT) / sqrts
    q2 = m_Z2 + pT * pT
```

I would invert these two steps so you can do

```
    ET2 = m_Z2 + pT * pT
    x = (np.sqrt(ET2) + pT) / sqrts
```

and return ET2 so that is clearer.

I'm not sure which computation of `x` is correct though, @comane where did you get yours from ?

Reply:
Just to follow up, this PR now uses the following definition
```
    ET2 = m_Z2 + pT * pT
    x = (np.sqrt(ET2) + pT) / sqrts ,
```
which has been taken from [here](https://github.com/NNPDF/nnpdf/blob/9b1e436f5a36541e88f6e45ea27fe44012efebc7/validphys2/src/validphys/plotoptions/kintransforms.py#L207).